test/com/google/enterprise/adaptor/filenet/FileNetAdaptorTest.java

```diff
@@ -73,4 +97,282 @@ public void testGetConnection() throws Exception {
+  }
+
+  @Test
+  public void testCheckpoint_ctor_threeArgs() throws Exception {
+    SimpleDateFormat dateFmt
+        = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    Date now = new Date();
+    String traverser = "document";
+    Id id = new Id("{AAAAAAAA-0000-0000-0000-000000000000}");
+
+    Checkpoint checkpoint = new Checkpoint(traverser, now, id);
+    assertEquals(traverser, checkpoint.type);
+    assertEquals(now, checkpoint.timestamp);
+    assertEquals(id, checkpoint.guid);
+  }
+
+  @Test
+  public void testCheckpoint_toString_shortCheckpoint() {
+    String checkpointStr = "type=document";
+    Checkpoint checkpoint = new Checkpoint(checkpointStr);
+    assertEquals(checkpointStr, checkpoint.toString());
+  }
+
+  @Test
+  public void testCheckpoint_toString_longCheckpoint() throws Exception {
+    SimpleDateFormat dateFmt
+        = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    Date now = new Date();
+    String checkpointStr = "type=document;timestamp=" + dateFmt.format(now)
+       + ";guid={AAAAAAAA-0000-0000-0000-000000000000}";
+    Checkpoint checkpoint = new Checkpoint(checkpointStr);
+    assertEquals(checkpointStr, checkpoint.toString());
+  }
+
+  // These tests of getQueryTimeString are adapted from the v3 FileUtilsTest.
+  private void testCheckpoint_getQueryTimeString(String tzStr,
+      String dateUnderTest, String expected) throws Exception {
+    TimeZone.setDefault(TimeZone.getTimeZone(tzStr));
+    Checkpoint checkpoint = new Checkpoint("type=test;timestamp="
+        + dateUnderTest + ";guid={AAAAAAAA-0000-0000-0000-000000000000}");
+    assertEquals(expected, checkpoint.getQueryTimeString());
+  }
+
+  @Test
+  public void testCheckpoint_getQueryTimeString_West() throws Exception {
+    testCheckpoint_getQueryTimeString("GMT-0800",
+        "2013-04-30T02:00:00.392-0800",
+        "2013-04-30T02:00:00.392-08:00");
+  }
+
+  @Test
+  public void testCheckpoint_getQueryTimeString_Utc() throws Exception {
+    testCheckpoint_getQueryTimeString("GMT",
+        "2013-04-30T10:00:00.392+0000",
+        "2013-04-30T10:00:00.392+00:00");
+  }
+
+  @Test
+  public void testCheckpoint_getQueryTimeString_East() throws Exception {
+    testCheckpoint_getQueryTimeString("GMT+0400",
+        "2013-04-30T14:00:00.392+0400",
+        "2013-04-30T14:00:00.392+04:00");
+  }
+
+  @Test
+  public void testInit_maxFeedUrlsTooSmall() throws Exception {
+    config.overrideKey("feed.maxUrls", "1");
+    thrown.expect(InvalidConfigurationException.class);
+    thrown.expectMessage("feed.maxUrls must be greater than 2");
+    adaptor.init(context);
+  }
+
+  @Test
+  public void testGetDocIds() throws Exception {
+    long startTime = new Date().getTime();
+    RecordingDocIdPusher pusher
+        = (RecordingDocIdPusher) context.getDocIdPusher();
+    config.overrideKey("feed.maxUrls", "5");
+    adaptor.init(context);
+    adaptor.getDocIds(pusher);
+
+    List<Record> golden = ImmutableList.of(
```

Comment:
I'm not a fan of "golden". Can we stick to "expected"? I'd also like to have a convention for the asserts. I'm not usually a fan of foo = ...; bar = ...; assertEquals(foo, bar). I prefer inlining with the exception of never inlining calls to the object under test. What preference have you?